app/src/main/java/org/simple/clinic/contactpatient/ContactPatientBottomSheet.kt

```diff
@@ -3,6 +3,8 @@ package org.simple.clinic.contactpatient
 import android.content.Context
```

Comment:
I have a question about why did we choose to flip views in the `ContactPatientBottomSheet` rather than opening the sheet separately. Is it to avoid the navigation issues between multiple bottom sheets ?
And are we are planning to do the same for `RemoveAppointmentSheet` as well?

Reply:
Hmm, this is weird. I remember adding the reasoning as a commit message, but it seems like I missed it.

The overall reasoning is correct, I did this to:

- Avoid navigation issues between multiple bottom sheets.
- Make migrating this set of screens to the navigation component easier. Once this is done, since nav component has support for bottom sheet fragments, we can split these screens out again. Mobius makes that pretty easy for us.
- In the final part of this PR, I change the overdue screen to open this screen itself and delete all the screens that are a part of the older call patient flow, so this screen is the only one necessary.

Reply:
Added this reasoning to the kdoc of `UiMode`.

Reply:
iirc, the issue with navigation occurs when we have to open a `Screen` from the `Sheet`. Since here we are opening different sheets from one sheet itself, I don't think it should be a problem. That will avoid the extra work of separating out the sheets later on as well.

Reply:
The problem will be present here because the final view, the `RemoveAppointment` view is technically a "screen" and not a bottom sheet.

I think I'd like to get this in as it is now, and then separate them out once we migrate to the navigation component properly since that will resolve a lot of navigation problems that we have.

Reply:
Oh. I forgot we changed `RemoveAppointment` to a screen.
Cool. Let's keep it this way.